### PR TITLE
[SCA] Minor improvement on Psalm typing

### PR DESCRIPTION
**Pull request by phansys**
|      Q       |   A
|------------- | -----------
| Type         | improvement
| Fixed issues | n/a

#### Summary

Spin-off from #5998, much less ambitious.

### CODE REVIEW

**Review by greg0ire — DISMISSED** [review]

File: src/Driver/SQLite3/Statement.php [hunk 1]
```diff
@@ -40,7 +40,13 @@ public function __construct(SQLite3 $connection, SQLite3Stmt $statement)
         $this->statement  = $statement;
     }
 
-    /** @inheritdoc */
+    /**
+     * @throws UnknownParameterType
+     *
+     * @inheritdoc
```
Comment (SenseException) [inline src/Driver/SQLite3/Statement.php]:
```suggestion
     * {@inheritdoc}
```

File: src/Driver/SQLite3/Statement.php [hunk 2]
```diff
@@ -55,7 +61,13 @@ public function bindValue($param, $value, $type = ParameterType::STRING): bool
         return $this->statement->bindValue($param, $value, $this->convertParamType($type));
     }
 
-    /** @inheritdoc */
+    /**
+     * @throws UnknownParameterType
+     *
+     * @inheritdoc
```
Comment (SenseException) [inline src/Driver/SQLite3/Statement.php]:
```suggestion
     * {@inheritdoc}
```

File: src/DriverManager.php [hunk 3]
```diff
@@ -96,7 +96,8 @@ final class DriverManager
      *
      * @deprecated Use actual driver names instead.
      *
-     * @var string[]
+     * @var array<string, string>
```
Comment (SenseException) [inline src/DriverManager.php:99]: Is this line redundant when `@psalm-var` is used? We don't generate docs from PHPDoc docblocks and the psalm annotation does the trick too.

**Review by SenseException — CHANGES_REQUESTED** [review]

File: src/Driver/SQLite3/Statement.php (hunk 1 above)
Comment (phansys) [reply src/Driver/SQLite3/Statement.php]: I could make a different PR updating this in the whole repo if you agree. There are much more occurrences.

File: src/DriverManager.php (hunk 3 above)
Comment (phansys) [reply src/DriverManager.php:99]: I supposed that native PHPDoc tags were kept in order to maintain compatibility with tools (IDEs, etc) that don't support Psalm extended typing.
If you think they are not required, I'm fine removing them.

File: src/Driver/SQLite3/Statement.php (hunk 1 above)
Comment (SenseException) [reply src/Driver/SQLite3/Statement.php]: It was correct when it was on its own, so it's fine when you're doing it only for the two occurrences of this PR.

File: src/DriverManager.php (hunk 3 above)
Comment (SenseException) [reply src/DriverManager.php:99]: I guess you have a point that some IDEs won't understand the psalm part. :+1:

**Review by SenseException — APPROVED** [review]

**Pull request merged**